### PR TITLE
Added a function to split a scalar into multiple scalars

### PR DESCRIPTION
**Pull request by JoaoAndreSa**
Easy peasy

``` go
// SplitScalar splits a given scalar into multiple n+1 scalars
func SplitScalar(rootScalar kyber.Scalar, nbrSplits int) []kyber.Scalar {
	allScalars := make([]kyber.Scalar, 0)
	for i:=0; i<nbrSplits; i++ {
		newScalar := RandomScalarSlice(1)[0]
		rootScalar = rootScalar.Sub(rootScalar, newScalar)
		allScalars = append(allScalars, newScalar)
	}
	allScalars = append(allScalars, rootScalar)

	return allScalars
}
```

### CODE REVIEW

File: lib/crypto.go [hunk 1]
```diff
@@ -112,6 +112,19 @@ func GenKeys(n int) (kyber.Point, []kyber.Scalar, []kyber.Point) {
 	return group, priv, pub
 }
 
+// SplitScalar splits a given scalar into multiple n+1 scalars
+func SplitScalar(rootScalar kyber.Scalar, nbrSplits int) []kyber.Scalar {
+	allScalars := make([]kyber.Scalar, 0)
```
Comment (ineiti) [inline lib/crypto.go]:
```suggestion
	allScalars := make([]kyber.Scalar, nbrSplits)
```

File: lib/crypto.go [hunk 2]
```diff
@@ -112,6 +112,19 @@ func GenKeys(n int) (kyber.Point, []kyber.Scalar, []kyber.Point) {
...
+	for i := 0; i < nbrSplits; i++ {
+		newScalar := RandomScalarSlice(1)[0]
```
Comment (ineiti) [inline lib/crypto.go]:
```suggestion
		allScalars[i] := RandomScalarSlice(1)[0]
```

File: lib/crypto.go [hunk 3]
```diff
@@ -112,6 +112,19 @@ func GenKeys(n int) (kyber.Point, []kyber.Scalar, []kyber.Point) {
...
+		rootScalar = rootScalar.Sub(rootScalar, newScalar)
```
Comment (ineiti) [inline lib/crypto.go]:
```suggestion
		rootScalar = rootScalar.Sub(rootScalar, allScalars[i])
```

File: lib/crypto.go [hunk 4]
```diff
@@ -112,6 +112,19 @@ func GenKeys(n int) (kyber.Point, []kyber.Scalar, []kyber.Point) {
...
+		allScalars = append(allScalars, newScalar)
```
Comment (ineiti) [inline lib/crypto.go]:
```suggestion
```

File: lib/crypto.go [hunk 5]
```diff
@@ -112,6 +112,19 @@ func GenKeys(n int) (kyber.Point, []kyber.Scalar, []kyber.Point) {
 	return group, priv, pub
 }
 
+// SplitScalar splits a given scalar into multiple n+1 scalars
+func SplitScalar(rootScalar kyber.Scalar, nbrSplits int) []kyber.Scalar {
+	allScalars := make([]kyber.Scalar, 0)
+	for i := 0; i < nbrSplits; i++ {
```
Comment (ineiti) [inline lib/crypto.go]:
```suggestion
	for i := range allScalars {
```

File: lib/crypto.go [hunk 6]
```diff
@@ -112,6 +112,19 @@ func GenKeys(n int) (kyber.Point, []kyber.Scalar, []kyber.Point) {
 	return group, priv, pub
 }
 
+// SplitScalar splits a given scalar into multiple n+1 scalars
```
Comment (ineiti) [inline lib/crypto.go]:
```
// The sum of the returned slice of scalars equals the given `rootScalar`
```

**Comment by ineiti** [comment]
Sorry, couldn't resist doing some goification...

File: lib/crypto.go (hunk 6 above)
Comment (JoaoAndreSa) [reply lib/crypto.go]: 2c1e469

**Comment by JoaoAndreSa** [comment]
Merci @ineiti. Feel free to goify things :D

**Pull request merged**